### ISSUE TITLE
A more permissive license

### ISSUE THREAD

**Issue by lbargaoanu** [ReflectionMagic/ReflectionMagic#29]
Would it be possible to switch to [the MIT license](https://en.wikipedia.org/wiki/MIT_License)? That seems to be more permissive than MSPL. Projects with an MIT license: Newtonsoft.Json, Nunit, AutoMapper, Autofac, .Net Core, jQuery. Apache 2.0 seems even more permissive, but I'm no lawyer.

**Comment by jvandertil** [comment]
Hello, while I'm not against switching licenses I do wonder what the problem is with the current license. Which parts of the license are restricting to you?

In any case I would have to defer this decision to @davidebbo as he is the original author of the library and released it under MS-PL.

**Comment by lbargaoanu** [comment]
I'd have to ask legal about that :) I see your point, but as far as I'm concerned a more permissive license is the easy way out. I know they have no issue with the MIT license.

**Comment by lbargaoanu** [comment]
I'm guessing technically it would take a new release for the new license to be in effect.

**Comment by jvandertil** [comment]
Yes, but I still have some bugs to solve before I get round to releasing a new version. 

Since I consider changing the license a breaking change, I will roll everything into 4.0.0. I hope I can get round to doing that (probably a beta) later this week.

**Comment by lbargaoanu** [comment]
Thanks.
